### PR TITLE
MINOR: refactor streams system test class hierachy

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
@enothereska @dguy @guozhangwang 

Reworking in preparation to add new system tests. 

Also for `0.10.2`

**Comment by guozhangwang** [comment]
LGTM overall, could you trigger one `simplebenchmark` test locally to make sure it works properly @mjsax ?

**Comment by mjsax** [comment]
I did test with `streams_shutdown_deadlock_test.py` and it works there. (Also with the new test to be added for client-broker backwards compatibility).

For `streams_simple_benchmark_test.py` I found one problem that I fixed. But the benchmark never complete locally but crashes and times out. As it starts up, I guess it is working fine.

The error I get in stderr is (seems to be unrelated to the changes though -- WDYT?):
````
Exception in thread "StreamThread-3" org.apache.kafka.streams.errors.StreamsException: stream-thread [StreamThread-3] Failed to rebalance
        at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:617)
        at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:369)
Caused by: java.lang.NullPointerException
        at org.apache.kafka.streams.processor.internals.ProcessorStateManager.register(ProcessorStateManager.java:133)
        at org.apache.kafka.streams.processor.internals.AbstractProcessorContext.register(AbstractProcessorContext.java:99)
        at org.apache.kafka.streams.state.internals.RocksDBStore.init(RocksDBStore.java:160)
        at org.apache.kafka.streams.state.internals.ChangeLoggingKeyValueBytesStore.init(ChangeLoggingKeyValueBytesStore.java:39)
        at org.apache.kafka.streams.state.internals.ChangeLoggingKeyValueStore.init(ChangeLoggingKeyValueStore.java:60)
        at org.apache.kafka.streams.state.internals.MeteredKeyValueStore$7.run(MeteredKeyValueStore.java:100)
        at org.apache.kafka.streams.processor.internals.StreamsMetricsImpl.measureLatencyNs(StreamsMetricsImpl.java:188)
        at org.apache.kafka.streams.state.internals.MeteredKeyValueStore.init(MeteredKeyValueStore.java:131)
        at org.apache.kafka.streams.processor.internals.AbstractTask.initializeStateStores(AbstractTask.java:86)
        at org.apache.kafka.streams.processor.internals.StreamTask.<init>(StreamTask.java:141)
        at org.apache.kafka.streams.processor.internals.StreamThread.createStreamTask(StreamThread.java:839)
        at org.apache.kafka.streams.processor.internals.StreamThread$TaskCreator.createTask(StreamThread.java:1212)
        at org.apache.kafka.streams.processor.internals.StreamThread$AbstractTaskCreator.retryWithBackoff(StreamThread.java:1185)
        at org.apache.kafka.streams.processor.internals.StreamThread.addStreamTasks(StreamThread.java:942)
        at org.apache.kafka.streams.processor.internals.StreamThread.access$500(StreamThread.java:69)
        at org.apache.kafka.streams.processor.internals.StreamThread$1.onPartitionsAssigned(StreamThread.java:236)
        at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.onJoinComplete(ConsumerCoordinator.java:230)
        at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:336)
        at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:300)
        at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:261)
        at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1025)
        at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:990)
        at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:587)
        ... 1 more
```

**Comment by enothereska** [comment]
Given that this changes the python code, it'd be good to trigger all system tests, including the smoke one and bounce one before merging.

**Comment by enothereska** [comment]
Also the benchmark test has always ran nightly, that needs to be part of tests to be run (as @mjsax has already done, thanks)

**Comment by enothereska** [comment]
@guozhangwang in an ideal world, this PR and PR: https://github.com/apache/kafka/pull/2391/files should go together.

**Comment by guozhangwang** [comment]
@enothereska System tests will take about 5 hours, and given that we are going to have nightly builds in another 10 hour, I'd say just watch the upcoming system test after merging this.

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk and 0.10.2.

**Comment by enothereska** [comment]
@guozhangwang I think this is really bad. System tests for streams take just 5 mins/test, you don't have to run all the system tests, just streams.

**Comment by ijuma** [comment]
I agree with @enothereska, a change like this should not be merged before running the Streams system tests via the branch builder. It doesn't take long since there are a small number of Streams tests.

For Core, whenever we change shared security code, we run all the tests before merging even though they take several hours as there is a lot of value in catching issues before the code is merged.

File: tests/kafkatest/services/performance/streams_performance.py [hunk 1]
```diff
@@ -13,115 +13,17 @@
...
-        self.logger.info("Starting SimpleBenchmark process on " + str(node.account))
-        results = {}
-        with node.account.monitor_log(self.STDOUT_FILE) as monitor:
-            node.account.ssh(self.start_cmd(node))
-            monitor.wait_until('SimpleBenchmark instance started', timeout_sec=15, err_msg="Never saw message indicating SimpleBenchmark finished startup on " + str(node.account))
-
-        if len(self.pids(node)) == 0:
-            raise RuntimeError("No process ids recorded")
-
-    def collect_data(self, node):
```
Comment (enothereska) [inline tests/kafkatest/services/performance/streams_performance.py:120]: Why is this removed?

Comment (mjsax) [reply tests/kafkatest/services/performance/streams_performance.py:120]: This did not happen intentionally... Will open a PR to fix it.

**Comment by guozhangwang** [comment]
@enothereska @ijuma My apologies, I mis-understood Eno's comment before and was careless for merging the PR.

File: tests/kafkatest/services/streams.py [hunk 2]
```diff
@@ -125,24 +128,35 @@ def start_node(self, node):
 
         node.account.create_file(self.LOG4J_CONFIG_FILE, self.render('tools_log4j.properties', log_file=self.LOG_FILE))
 
-        self.logger.info("Starting StreamsSmokeTest process on " + str(node.account))
+        self.logger.info("Starting StreamsTest process on " + str(node.account))
         with node.account.monitor_log(self.STDOUT_FILE) as monitor:
             node.account.ssh(self.start_cmd(node))
-            monitor.wait_until('StreamsSmokeTest instance started', timeout_sec=15, err_msg="Never saw message indicating StreamsSmokeTest finished startup on " + str(node.account))
+            monitor.wait_until('StreamsTest instance started', timeout_sec=60, err_msg="Never saw message indicating StreamsTest finished startup on " + str(node.account))
```
Comment (enothereska) [inline tests/kafkatest/services/streams.py:134]: This probably doesn't need to be increased since they start fast.

Comment (mjsax) [reply tests/kafkatest/services/streams.py:134]: Ack.
